src/pages/Home.tsx

```diff
@@ -1,22 +1,236 @@
+        <img
+          src={currentActiveUser ? currentActiveUser.bgImage : coverNone}
+          alt="background"
+          className="h-full w-full"
+        />
+      </div>
+      <div className="mt-5 absolute left-40 top-80 object-cover">
+        <img
+          src={currentActiveUser ? currentActiveUser.avatar : avatarNone}
+          alt="avatar"
+          className="w-40 h-40 rounded-full"
+        />
+      </div>
+
+      <div className="p-5 flex justify-center items-center flex-col">
+        <div>
+          <Button
+            background={false}
+            onClick={() => {
+              setMyTweets(false);
+              setIsShowAllTweets(!isShowAllTweets);
+            }}
+            disabled={isShowAllTweets || !currentActiveUser}
+          >
+            All Tweets
+          </Button>
+
+          <Button
+            background={false}
+            onClick={() => {
+              setIsShowAllTweets(false);
+              setMyTweets(!myTweets);
+            }}
+            disabled={myTweets || !currentActiveUser}
+          >
+            My Tweets
+          </Button>
+        </div>
+        <p className="italic hover: pl-32">
+          {currentActiveUser && currentActiveUser.tweets.length}
+        </p>
+      </div>
+
+      <div className="flex flex-row">
+        <div className="w-1/4 h-96 m-5 flex flex-col justify-center items-center">
+          <p className="font-bold m-4 text-xl">Personal info</p>
+          <p className="font-bold italic hover:not-italic">
+            {currentActiveUser
+              ? `${currentActiveUser.firstName} ${currentActiveUser.lastName}`
+              : 'Surname name ...'}
+          </p>
+          <p className="italic hover:not-italic">
+            {currentActiveUser ? currentActiveUser.location : 'Location ...'}
+          </p>
+          <p className="italic hover:not-italic pb-10">
+            {currentActiveUser ? currentActiveUser.joined : 'Joined at ...'}
+          </p>
+          <Button
+            background={false}
+            onClick={logOut}
+            disabled={!currentActiveUser}
+          >
+            Log out
+          </Button>
+        </div>
+
+        <div className="w-2/3 m-10">
+          {isShowAllTweets &&
+            users.map((user: IUser) =>
+              user.tweets.map((item: ITweet, index: number) => (
+                <TweetContainer
+                  avatarTweet={user.avatar || avatarNone}
+                  key={index}
+                  tweet={item.text}
+                  count={item.likes}
+                  nickName={user.username}
+                  name={user.firstName}
+                  date={item.date}
+                  currentUserId={currentActiveUser.id}
+                />
+              ))
+            )}
+
+          {myTweets &&
+            currentActiveUser &&
+            currentActiveUser.tweets.map(
+              (singleTweet: ITweet, index: number) =>
+                singleTweet && (
```

Comment:
why we check here for singleTweet? maybe we can just check if there's any items in array? currentActiveUser.length && ... for example